libsnark/zk_proof_systems/plonk/tests/test_plonk.cpp

```diff
@@ -1087,6 +1095,124 @@ template<typename ppT> void test_plonk_gates_matrix_transpose()
     ASSERT_EQ(gates_matrix_transpose, example.gates_matrix_transpose);
 }
 
+// generic test for all curves
+template<typename ppT> void test_plonk_constants_k1_k2()
+{
+    ppT::init_public_params();
+
+    using Field = libff::Fr<ppT>;
+    Field k1, k2;
+    plonk_generate_constants_k1_k2(k1, k2);
+    ASSERT_TRUE(plonk_are_valid_constants_k1_k2(k1, k2));
+}
+
+// generic test for all curves for an alternative method to generate
+// constants k1,k2 using randomization
+template<typename ppT> void test_plonk_random_constants_k1_k2()
+{
+    ppT::init_public_params();
+
+    using Field = libff::Fr<ppT>;
+    Field k1, k2;
+    plonk_generate_random_constants_k1_k2(k1, k2);
+    ASSERT_TRUE(plonk_are_valid_constants_k1_k2(k1, k2));
+}
+
+// test specific to BLS12-381 (uses class example specific to
+// BLS12-381)
+template<typename ppT> void test_plonk_constants_k1_k2_bls12_381()
+{
+    ppT::init_public_params();
+
+    using Field = libff::Fr<ppT>;
+    Field k1, k2;
+
+    // Check that the example k1 and k2 are valid (according to
+    // plonk_are_valid_constants_k1_k2)"
```

Comment:
```suggestion
    // plonk_are_valid_constants_k1_k2)".
```
And similarly elsewhere please.

Reply:
Is the suggestion here to add a full stop at the end of each comment? Besides, as another correction I think that the quote " at the end is redundant.

Reply:
> Is the suggestion here to add a full stop at the end of each comment?

Yes pleae, where comments are sentences.  Don't need to go through the entire plonk code, but just where you notice.

> Besides, as another correction I think that the quote " at the end is redundant.

Oh yes, I hadn't spotted that.  Thanks.

Reply:
Fixed.